ext/opcache/jit/zend_jit_helpers.c

```diff
@@ -500,6 +500,31 @@ static void ZEND_FASTCALL zend_jit_fetch_dim_r_helper(zend_array *ht, zval *dim,
 			}
 			ZEND_FALLTHROUGH;
 		case IS_NULL:
+			/* The array may be destroyed while throwing the notice.
```

Comment:
Please, no.
I would prefer if you just fetched the result first, copied it to the return value zval, and then emitted the deprecation. Similarly in many other places (this won't be possible for _all_ code because some return a `zval*` to the array slot).

Reply:
In what way did it fail?

Reply:
Probably because you don't emit the warning when `!retval` is true?